addons/base_iban/static/src/components/iban_widget/iban_widget.js

```diff
@@ -0,0 +1,45 @@
+/** @odoo-module **/
+
+import { registry } from "@web/core/registry";
+import { CharField } from "@web/views/fields/char/char_field";
+import { useDebounced } from "@web/core/utils/timing";
+import { useService } from "@web/core/utils/hooks";
+
+const { useState, useEffect } = owl;
+
+export class IbanWidget extends CharField {
```

Comment:
what features of the charfield are you using?

Reply:
Short story:
If you don't put a widget, the Charfield would be the one called.
As I just want to add some behavior to it, I extended it.

Long story:
Not putting it force me to deals with several problems. 
- I have to first use `useInputField` to prevent the input to be wipe out.
- I'm dealing with some weird visual issue (the input box gets its border being bold and the corners rounded)
- I'm dealing with issues with the field that is next by it: it says it's required and doesn't save even when I set it

Extending Charfield solves ALL of that and I'm sure it also deals with other stuff I haven't any idea about.